libyul/optimiser/StackToMemoryMover.cpp

```diff
@@ -98,68 +117,160 @@ m_nameDispenser(_context.dispenser)
 
 void StackToMemoryMover::operator()(FunctionDefinition& _functionDefinition)
 {
-	for (TypedName const& param: _functionDefinition.parameters + _functionDefinition.returnVariables)
-		if (m_memoryOffsetTracker(param.name))
-		{
-			// TODO: we cannot handle function parameters yet.
-			return;
-		}
+	// It is important to first visit the function body, so that it doesn't replace the memory inits for
+	// variable arguments we might generate below.
 	ASTModifier::operator()(_functionDefinition);
+
+	vector<Statement> memoryVariableInits;
+
+	// All function arguments with a memory slot are moved at the beginning of the function body.
```

Comment:
parameters

Reply:
I've always found our use of "argument" vs "parameter" a bit weird. Since we call return values "return parameters", I tend to call the actual parameters arguments... but I can change it to whatever we prefer, but I think calling return values "return parameters" in the AST is unfortunate.

Reply:
I think we settled with "return variables" instead of "return parameters".
So it's:
function call: argument and return value
function definition: parameter and return variable